### PR TITLE
[Merged by Bors] - Fix crash when using Duration::MAX

### PR DESCRIPTION
**Pull request by SUPERCILEX**
# Objective

If you set the `ReactiveLowPower` max wait to `Duration::MAX`, stuff panics. Fix that.

## Solution

Wait forever if addition failed.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Good choice. Can we document this behavior under `ControlFlow`?

**Comment by alice-i-cecile** [comment]
Ah my bad. No, I meant as a comment on the variants in [UpdateMode](https://dev-docs.bevyengine.org/bevy/winit/enum.UpdateMode.html#variant.Reactive).

**Comment by SUPERCILEX** [comment]
Gotya, done.

**Review by james7132 — APPROVED** [review]

File: crates/bevy_winit/src/winit_config.rs [hunk 1]
```diff
@@ -68,7 +68,11 @@ pub enum UpdateMode {
     /// Once the app has executed all bevy systems and reaches the end of the event loop, there is
     /// no way to force the app to wake and update again, unless a `winit` event (such as user
     /// input, or the window being resized) is received or the time limit is reached.
-    Reactive { max_wait: Duration },
+    Reactive {
+        /// The maximum time to wait before the event loop runs again. Note that if the duration is
```
Comment (alice-i-cecile) [inline crates/bevy_winit/src/winit_config.rs:72]: These should be line split, and the advice should be more precise. What do we need to set the value to in order to get an indefinite wait? 

In many ways, this is actually a feature, so we should demonstrate how to use it.

Comment (SUPERCILEX) [reply crates/bevy_winit/src/winit_config.rs]: Well the problem is that it's platform dependent. Some platforms store Instants as u32s, others store them as u64s so the answer is "dunno, you figure it out." :) That said, using Duration::MAX is effectively the same as waiting for ever so I made a note of that.

**Review by alice-i-cecile — APPROVED** [review]
Perfect, thanks!

**Pull request closed without merging**